### DISCUSSION TITLE
Wiring diagram for Auto Square/Ganged Y axis.

### DISCUSSION THREAD

**Discussion by PNC-Big-red** [phil-barrett/grblHAL-teensy-4.x#49] [Q&A] [answered]
Hi all, I would like to add auto-squaring and i have flashed grblHAL_T41 4X gangY AutoSq qei SDCard Keypad firmware.

Would I wire the limit switch on the Y2 gantry to the A- limit switch port on the breakout board? Is there any other configuring that I will need to do after wiring apart from enabling hard limits and homing?

With regards to the ganged-Y setup, i am guessing that Y2 will connect to the A-Port and my rotary Axis will hook up to the B-axis port?

Please forgive my ignorance. I am quite new to this!

Cheers!

**Comment by phil-barrett** [comment, marked as answer]
Have you read this [blog about autosquaring](https://www.grbl.org/single-post/it-s-hip-to-be-auto-square)? 

grblHAL assigns the ganged motor to the next available axis which is B in your case. So, you would use the B motor and B limit for the ganged motor and limit sensor.  grblHAL assigns the ganged motor to the next available axis which is B in your case. The A axis would go to your rotary.

**Reply by PNC-Big-red** [reply]
Thanks. I will try that. I completely missed the blog.

**Comment by AndyCXL** [comment]
Can I ask a follow-on question on this topic... Teensy 4.1/Phil-Barrett BOB/DM3230 drivers

I have XYZYA axis configuration, dual Y with independent drivers and steppers plus a rotary A axis. I want to implement auto-squaring on Y and the question of to which axis input does the Y2 inductive sensor connect.

As the driver for Y2 connects to the 4th 'port', I presume the Y2 homing sensor also connects to the 4th input. Correct?

Does the homing sequence need to be altered? Currently the machine homes as Z then XY together. Will auto-square need Z then X then Y, or can it manage Z then XY whilst auto-squaring?

Very happy with my upgrade - the base machine is (was) ShapeOKO 3XL, home-brew rotary along the X axis, gSender and a combination of VCarve Pro and Fusion 360 CAD/CAM. Looking forward to gSender getting DRO to support the rotary axis.

**Comment by phil-barrett** [comment]
In the case of a 4 axis machine, grblHAL gangs the B axis as Y2. No change
to the homing sequence.  The B sensor associates with with Y2. The 4th axis
is A.

**Reply by AndyCXL** [reply]
Phil - thank you for coming back on this. OK, so XYZAY is the sequence and I can hook up the Y2 sensor accordingly. I look forward to making this tweak to my machine.
Cheeky extra question... I use gSender from Sienci as my sender app, and it looks for 'Grbl 1.1' or similar as a controller boot response. I started out with Compatibility level 0 have been issuing &RST=& manually to trigger this response for gSender, but it's a bit of a hassle. I don't believe Axis auto-squaring would be affected at all if I were to switch to Compatibility level 1, but to know this for a fact would be helpful.

**Comment by phil-barrett** [comment]
I don't know the answer to that.  You could put the trigger sequence in a
macro? ioSender doesn't have that issue.

**Reply by AndyCXL** [reply]
Phil - no worries, I can play around with this and see. My workflow is all Mac based, so iOSender whilst interesting falls outside of this unless/until a native Mac option appears.
Thanks for your feedback.

**Comment by phil-barrett** [comment]
Ah, my bad. (I wish ioSender wasn't written in C# but it wasn't my call!)
